### DISCUSSION TITLE
RAC Table with option to render as other element types

### DISCUSSION THREAD

**Discussion by splicebison** [adobe/react-spectrum#6939] [Ideas]
Hi all, I wanted to start the discussion and get feedback about possibly implementing the RAC table with an option to render other element types instead of the semantic table. This would allow for a more Spectrum-like design pattern (TableView). I realize this could be implemented with hooks but would be happy to see it go into core RAC.

Currently, `useElementType` in Table has hard-coded values. I haven't parsed the source code thoroughly but am wondering if it's possible to easily allow (with context) the ability to optional render divs while still maintaining accessibility and order.

**Comment by snowystinger** [comment]
We're currently working on a Spectrum 2 TableView based on RAC here https://github.com/adobe/react-spectrum/pull/6778

**Reply by splicebison** [reply]
Hi @snowystinger, thanks for the response. I had a pass at the PR as well as the Storybook preview build in the PR and maybe I am missing something but the implementation on the Spectrum table views still seems to be with hooks and rendered `div` elements while the underlying RAC component is still semantic table elements. Will this be changing with the upcoming v2 release of Spectrum or will the RAC component stay with table elements? I ask because I would prefer using the RAC component in our design system but if there are no current plans for changing this, I might go ahead and implement the hook-based tableview using react-aria instead.

Also, if you take outside contributions, it's something I could potentially work on (the RAC implementation) but wanted to start a discussion first to see if it's of interest to the team or not.

**Comment by splicebison** [comment]
After another review of the PR and codebase (along with releases) I found the undocumented `UNSTABLE_TableLayout` and `UNSTABLE_Virtualizer` that allows for render a virtualized tableview similar to Spectrum and using `div` elements. This is already available in the summer release. This is absolutely fantastic! I will be using this moving forward in anticipation of the stable release into RAC. Thanks for pointing me in the right direction. I am going to close this discussion now.
